src/frameworks/ccpa_from_us_privacy_string.js

```diff
@@ -0,0 +1,47 @@
+const FrameworkBase = require('./base');
+
+/**
+ * Implements usprivacy string framework
+ * for more information on the US Privacy string see:
+ * https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md#us-privacy-string-format
+ */
+class CCPAFromUSPrivacyString extends FrameworkBase {
+  constructor() {
+    super();
+    this.usPrivacyString = '';
+  }
+
+  useConfig({ usp }) {
+    if (usp) {
+      this.usPrivacyString = ('' + usp).toUpperCase();
+    }
+  }
+
+  supportedCapabilities() {
+    return ['canUsePersonalInformationForTargeting', 'hasBeenNotifiedOfRights'];
+  }
+
+  canUsePersonalInformationForTargeting() {
+    return this.consentStringAllowsPersonalDataSale();
+  }
+
+  hasBeenNotifiedOfRights() {
```

Comment:
_Not-blocking/for-the-future:_ I don't think of `hasBeenNotifiedOfRights` as something the framework should be _exposing_. The engineer using this library doesn't actually care about that; they care about what they can *do*; I think this is an internal implementation detail.

Reply:
Yeah I debated this a bunch as well. I struggled to model the "capability" interface, and I found this "capability"  would helpful to solve: **Decisioning on if we need to present the user with a new window/dialog that should show them they have new rights**. (Which is user facing)

I suspect this will be modeled better in the future though, since it is clunky now.

Reply:
Ooh, I was not expecting a pushback that I would instantly agree with, but I think that's a _great_ point. Consider my comment retracted. 🙇  👍